pkgs/stdenv/generic/setup.sh

```diff
@@ -1,11 +1,12 @@
+# shellcheck disable=SC2148
```

Comment:
It would be good to add an explanation attached to the disable. That way, future readers can know why without digging in the git history.

Reply:
You mean a line under referencing the shellcheck for example?

Reply:
```suggestion
# shellcheck shell=bash
```

Reply:
I've just added a commit that adds a link for the shellcheck disable in setup.sh
I don't want to go overboard on this yet, as I'm afraid this is going to get harder to merge.
I personally like having a link, it's much more outsider friendly.